provider.go

```diff
@@ -172,21 +167,20 @@ func GetCredential(secrets map[string]string) (string, string, error) {
 }
 
 func (p *Provider) getVaultURL(ctx context.Context, cloudName string) (vaultURL *string, err error) {
-	log.Debugf("subscriptionID: %s", p.SubscriptionID)
 	log.Debugf("vaultName: %s", p.KeyvaultName)
-	log.Debugf("resourceGroup: %s", p.ResourceGroup)
 
-	vaultsClient := kvmgmt.NewVaultsClient(p.SubscriptionID)
-	token, tokenErr := p.GetManagementToken(AuthGrantType(), cloudName)
-	if tokenErr != nil {
-		return nil, errors.Wrapf(tokenErr, "failed to get management token")
+	// See docs for validation spec: https://docs.microsoft.com/en-us/azure/key-vault/about-keys-secrets-and-certificates#objects-identifiers-and-versioning
+	if match, _ := regexp.MatchString("[-a-zA-Z0-9]{3,24}", p.KeyvaultName); !match {
+		return nil, errors.Errorf("Invalid vault name: %q, must match [-a-zA-Z0-9]{3,24}", p.KeyvaultName)
 	}
-	vaultsClient.Authorizer = token
-	vault, err := vaultsClient.Get(ctx, p.ResourceGroup, p.KeyvaultName)
+
+	vaultDnsSuffix, err := GetVaultDNSSuffix(cloudName)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get vault %s", p.KeyvaultName)
+		return nil, err
 	}
-	return vault.Properties.VaultURI, nil
+	vaultDnsSuffixValue := *vaultDnsSuffix
```

Comment:
`VaultURI` does have the value. But with this change, instead of getting the vault and then the `VaultURI` from that, we are constructing the `VaultURI` by just using the vault name and the cloud name.